core/src/main/scala/org/apache/spark/deploy/PythonRunner.scala

```diff
@@ -82,7 +82,7 @@ object PythonRunner {
         s"spark-submit is currently only supported for local files: $path")
     }
     val windows = Utils.isWindows || testWindows
-    var formattedPath = if (windows) Utils.formatWindowsPath(path) else path
+    var formattedPath = Utils.formatPath(path, windows)
```

Comment:
This is the only use of `formatPath` leftover and it doesn't feel necessary. Instead, how about:

    val formattedPath = Try(new URI(formattedPath).getPath()).getOrElse(formattedPath)

Then you can remove `Utils.formatPath`. Maybe even `Utils.windowsDrive`.